fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/source/enumerator/FlinkSourceEnumerator.java

```diff
@@ -263,8 +264,13 @@ private List<SourceSplitBase> initNonPartitionedSplits() {
 
     private Set<PartitionInfo> listPartitions() {
         try {
-            List<PartitionInfo> partitionInfos = flussAdmin.listPartitionInfos(tablePath).get();
-            partitionInfos = applyPartitionFilter(partitionInfos);
+            PartitionSpec partitionSpec = convertFiltersToPartitionSpec(partitionFilters);
```

Comment:
We can convert the filter to `partialPartitionSpec` during the construction of `FlinkSourceEnumerator` to avoid the repeated conversion for `listPartitions`.